plugin/geoip/geoip.go

```diff
@@ -0,0 +1,108 @@
+// Package geoip implements a max mind database plugin.
+package geoip
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+	"github.com/oschwald/geoip2-golang"
+)
+
+var log = clog.NewWithPlugin(pluginName)
+
+// GeoIP is a plugin that add geo location data to the request context by looking up a maxmind
+// geoIP2 database, and which data can be later consumed by other middlewares.
+type GeoIP struct {
+	Next          plugin.Handler
+	db            db
+	langs         []string
+}
+
+type db struct {
+	*geoip2.Reader
+	// provides defines the schemas that can be obtained by querying this database, by using
+	// bitwise operations.
+	provides int
+}
+
+const (
+	anonymousIP = 1 << iota
+	asn
+	city
+	connectionType
+	country
+	domain
+	enterprise
+	isp
+)
+
+// probingIP is an arbitrary internet IP used to discover database types through database lookups.
+var probingIP = net.ParseIP("81.2.69.142")
```

Comment:
this is a weird thing to have, please remove

Reply:
The plugin validates that a `mmdb` database file provides the right schema by querying up the database then looking into the error returned to see if the schema is valid.

To do that the plugin is leveraging reader.City method whose signature requires a net.IP, we have to pass any valid net.IP value.

Would you be happy by changing the variable name and explaining a bit better why do we have that variable? do you propose to do the schema validation in a different way?